### DISCUSSION TITLE
Is there a way to detect if the task instance is being run from the ui rather than from upstream?

### DISCUSSION THREAD

**Discussion by p-hebert** [apache/airflow#23101] [Q&A] [answered]
Sister question: #23098

I created an operator that can be skipped based on a list of tags passed in the DagRun conf.
I would like to be able to override that skip behaviour when a user uses the UI to force re-run the skipped task so that I can run skipped tasks in development without having to re-run the dag.

Is there a way to detect if the task instance is being run from the ui rather than from upstream?
After spending 30 min inspecting both TaskInstance and Task from the template context I wasn't able to find anything useful to determine that.

**Comment by potiuk** [comment, marked as answer]
See https://github.com/apache/airflow/pull/22976 and https://github.com/apache/airflow/issues/21252 - I think we need that feature to distinguish manual operations and hopefully it will be added soon.

**Reply by p-hebert** [reply]
Thanks @potiuk! That's exactly what I was looking for :)
Btw, does the airflow contributors/users have any interest in a skippable-via-dagrun-conf operator? I would be happy to contribute my operator or a refactored version of it.
You can find the current impl in this gist here: https://gist.github.com/philippefutureboy/32a2853e3fffaa2b6faacd677176f786

**Reply by potiuk** [reply]
No idea. You can propose it always in a form of PR.

**Reply by p-hebert** [reply]
Will do!
